digdag-standards/src/main/java/io/digdag/standards/operator/redshift/BaseRedshiftLoadOperator.java

```diff
@@ -0,0 +1,230 @@
+            statusTableCleanupDuration = params.get("status_table_cleanup", DurationParam.class,
+                    DurationParam.of(Duration.ofHours(24)));
+        }
+        else {
+            statusTableName = null;
+            statusTableCleanupDuration = null;
+        }
+
+        UUID queryId;
+        // generate query id
+        if (!state.has(QUERY_ID)) {
+            // this is the first execution of this task
+            logger.debug("Generating query id for a new {} task", type());
+            queryId = UUID.randomUUID();
+            state.set(QUERY_ID, queryId);
+            throw TaskExecutionException.ofNextPolling(0, ConfigElement.copyOf(state));
+        }
+        queryId = state.get(QUERY_ID, UUID.class);
+
+        AWSCredentials baseCredentials = createBaseCredential(context.getSecrets());
+        AWSSessionCredentials sessionCredentials = createSessionCredentials(params, baseCredentials);
+        T statementConfig = createStatementConfig(params, sessionCredentials, queryId.toString());
+
+        beforeConnect(baseCredentials, statementConfig);
+
+        try (RedshiftConnection connection = connect(connectionConfig)) {
+            String query = buildSQLStatement(connection, statementConfig);
+
+            Exception statementError = connection.validateStatement(query);
+            if (statementError != null) {
+                statementConfig.accessKeyId = "********";
+                statementConfig.secretAccessKey = "********";
+                String queryForLogging = buildSQLStatement(connection, statementConfig);
+                throw new ConfigException("Given query is invalid: " + queryForLogging, statementError);
+            }
+
+            TransactionHelper txHelper;
+            if (strictTransaction) {
+                txHelper = connection.getStrictTransactionHelper(statusTableName,
+                        statusTableCleanupDuration.getDuration());
+            }
+            else {
+                txHelper = new NoTransactionHelper();
+            }
+
+            txHelper.prepare(queryId);
+
+            boolean executed = txHelper.lockedTransaction(queryId, () -> {
+                connection.executeUpdate(query);
+            });
+
+            if (!executed) {
+                logger.debug("Query is already completed according to status table. Skipping statement execution.");
+            }
+
+            try {
+                txHelper.cleanup();
+            }
+            catch (Exception ex) {
+                logger.warn("Error during cleaning up status table. Ignoring.", ex);
+            }
+
+            return TaskResult.defaultBuilder(request).build();
+        }
+        catch (LockConflictException ex) {
+            int pollingInterval = state.get(POLL_INTERVAL, Integer.class, INITIAL_POLL_INTERVAL);
+            // Set next interval for exponential backoff
+            state.set(POLL_INTERVAL, Math.min(pollingInterval * 2, MAX_POLL_INTERVAL));
```

Comment:
To be consistent with `td*` operators, you can include `poll_iteration` (number of iterations to calculate interval) instead of `poll_interval` (actually calculated interval). I think storing iteration is better to be able to support more options in the future such as td operator's min_poll_interval and max_poll_interval.
Or can't we use io.digdag.standards.operator.state.PollingRetryExecutor here with RuntimeException and DatabaseException as an idempotent exception?